backend/src/Notifo.Domain.Integrations/Discord/DiscordBotClientPool.cs

```diff
@@ -0,0 +1,41 @@
+﻿// ==========================================================================
+//  Notifo.io
+// ==========================================================================
+//  Copyright (c) Sebastian Stehle
+//  All rights reserved. Licensed under the MIT license.
+// ==========================================================================
+
+using Discord;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Notifo.Domain.Integrations.Discord;
+
+public class DiscordBotClientPool : CachePool<IDiscordClient>
+{
+    public DiscordBotClientPool(IMemoryCache memoryCache)
+        : base(memoryCache)
+    {
+    }
+
+    public async Task<IDiscordClient> GetDiscordClient(string botToken, CancellationToken ct)
+    {
+        var cacheKey = $"{nameof(IDiscordClient)}_{botToken}";
+
+        var found = await GetOrCreateAsync(cacheKey, TimeSpan.FromMinutes(5), async () =>
+        {
+            var client = new DiscordClient();
+
+            // Sadly it can't receive the cancellation token.
```

Comment:
No need to write that much, nobody reads it.  Just 

"Method provides no option to pass CancellationToken" is enough